app/src/main/java/com/example/plantsapp/domain/repository/TasksRepository.kt

```diff
@@ -2,8 +2,10 @@ package com.example.plantsapp.domain.repository
 
 import com.example.plantsapp.domain.model.Plant
 import com.example.plantsapp.domain.model.Task
+import java.util.Date
 
 interface TasksRepository {
     suspend fun addTasks(plant: Plant, tasks: List<Task>)
     suspend fun getTasksForPlant(plant: Plant): List<Task>
+    suspend fun updateTask(task: Task, completionDate: Date)
```

Comment:
There are 2 options of how it could be implemented: 
1. Your task contains date, so you can just create copy of Task with new date, and just make this method like
`suspend fun updateTask(task: Task)`, but you will need to create task everywhere where you use this updateTask method
or
2. You can leave like this, and create new RoomTask inside of implementation and leave message as it. I think that for current case it is okay to leave like this. So leave it in current implementation